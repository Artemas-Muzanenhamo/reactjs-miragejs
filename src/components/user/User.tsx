import React, {Component, ReactElement} from "react";

interface IUser {
    name: string;
    surname: string;
}

export class User extends Component<IUser> {
    render(): ReactElement<IUser> {
        return <div>
            <div className="uk-card uk-card-primary uk-card-body">
                <h3 className="uk-card-title">{this.props.name} {this.props.surname}</h3>
                <p>This is a profile card for {this.props.name}</p>
            </div>
        </div>
    }
}
