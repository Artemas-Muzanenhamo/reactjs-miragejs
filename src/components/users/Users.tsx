import React, {Component} from "react";
import {GetUser} from "../user/get_user/GetUser";
import {UsersApi} from "../../api/users/UsersApi";

interface IUser {
    name: string;
    surname: string;
}

export interface IUserJson {
    name: string
    surname: string
}

interface IUsers {
    users: IUser[];
}

export class Users extends Component {

    state: IUsers = {
        users: [{
            name: '',
            surname: ''
        }]
    };

    private mapToUser = (users: IUserJson[]): IUser[] => users
        .map(user => ({
                name: user.name,
                surname: user.surname
            })
        );

    componentDidMount(): void {
        UsersApi.fetchUsers()
            .then(this.mapToUser)
            .then(users => this.setState({users}));
    }

    render(): React.ReactElement {
        return <div className="uk-child-width-1-3@m uk-grid-small uk-grid-match uk-grid">
            {this.state.users
                .map(user =>
                    <GetUser key={user.name} name={user.name} surname={user.surname}/>
                )
            }
        </div>
    }
}
