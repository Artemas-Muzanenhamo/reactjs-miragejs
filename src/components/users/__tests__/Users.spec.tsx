import {shallow} from "enzyme";
import {User} from "../../user/User";
import React from "react";
import {IUserJson, Users} from "../Users";
import renderer from 'react-test-renderer';

describe('<Users />', () => {
    it('renders <User /> component', () => {
        let wrapper = shallow(<Users/>);
        expect(wrapper.find(User)).toHaveLength(1);
    });

    function testUsersJson(): IUserJson[] {
        return [
            {
                name: 'Dummy',
                surname: 'Test'
            },
            {
                name: 'Hello',
                surname: 'World'
            }
        ];
    }

    function mockSuccessFetchPromise(users = testUsersJson()) {
        return Promise.resolve({
            json: () => Promise.resolve(users),
            status: 200
        });
    }

    it('renders fetched Users', () => {
        jest.spyOn(global, 'fetch').mockImplementation(() => mockSuccessFetchPromise() as Promise<Response>);

        let tree = renderer.create(
            <Users/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
