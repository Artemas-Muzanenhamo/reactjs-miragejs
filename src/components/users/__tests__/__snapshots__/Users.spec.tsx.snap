// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Users /> renders fetched Users 1`] = `
<div
  className="uk-child-width-1-3@m uk-grid-small uk-grid-match uk-grid"
>
  <div>
    <div
      className="uk-card uk-card-primary uk-card-body"
    >
      <h3
        className="uk-card-title"
      >
        
         
        
      </h3>
      <p>
        This is a profile card for 
        
      </p>
    </div>
  </div>
</div>
`;
